Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectBuilder/MSBuildProjectReference.cs

```diff
@@ -25,16 +25,12 @@ public sealed partial class MSBuildProjectReference : ScriptableObject
         [Tooltip("The MSBuild build engine to use to build the project.")]
         private BuildEngine buildEngine = BuildEngine.DotNet;
 
-        [SerializeField]
-        [Tooltip("Indicates whether the referenced MSBuild project should automatically be built.")]
-        private bool autoBuild = true;
-
         [SerializeField]
         [Tooltip("Named profiles to configure different build options.")]
         private MSBuildBuildProfile[] profiles = new[]
         {
-            MSBuildBuildProfile.Create("Build", "-t:Build -p:Configuration=Release"),
-            MSBuildBuildProfile.Create("Rebuild", "-t:Rebuild -p:Configuration=Release"),
+            MSBuildBuildProfile.Create(name: "Build", autoBuild: true, arguments: "-t:Build -p:Configuration=Release"),
+            MSBuildBuildProfile.Create(name: "Rebuild", autoBuild: false, arguments: "-t:Rebuild -p:Configuration=Release"),
```

Comment:
Thinking about it a bit more. Should the primary default profile actually not specify a target argument. Then it just relies on the default behavior that the project was configured with. For most projects the default target will be Build, but then it would still work for projects that customized themselves to have a different default target.

Reply:
I thought about doing that, but there are already other assumptions in the profiles. For example, we also provide `Rebuild`, and both set `Configuration` to Release. I feel like we should either assume it's a standard project, or assume it is not a standard project, and if we assume it is not a default project we should be assuming `Rebuild` and `Configuration`. I think standard projects would be the common case, so I'm inclined to optimize for that (which I think is the same thing you were suggesting yesterday).